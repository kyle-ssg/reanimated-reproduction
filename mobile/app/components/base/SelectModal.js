import React, { Component } from "react";
import propTypes from "prop-types";
import { Navigation } from "react-native-navigation";
import { Loader } from "./Loader";

// propTypes: uri: RequiredString
const NativeModal = class extends Component {
  static displayName = "SelectModal";

  static propTypes = {
    componentId: propTypes.string,
    value: propTypes.object,
    isLoading: propTypes.bool,
    multiple: propTypes.bool,
    items: propTypes.any,
    onChange: propTypes.func.isRequired,
    placeholder: propTypes.string,
    filterItem: propTypes.func,
    renderRow: propTypes.func,
    autoclose: propTypes.bool,
  };

  constructor(props) {
    super(props); // todo:
    this.navigationEventListener = Navigation.events().bindComponent(this);
    this.state = {
      isLoading: true,
      value: props.value,
    };
  }

  onDone = () => {
    //todo: dismiss
    this.props.onChange(this.state.value);
  };

  onChange = (value) => {
    this.setState({ value }, this.props.autoclose && this.onDone);
  };

  render() {
    const {
      isLoading,
      multiple,
      items,
      placeholder,
      filterItem,
      renderRow,
    } = this.props;
    const { onChange } = this;
    const { value } = this.state;
    return (
        <Flex style={[Styles.body]}>
            {isLoading && (
            <Flex style={Styles.centeredContainer}>
                <Loader />
            </Flex>
        )}
            {items && (
            <Fade style={{ flex: 1 }} autostart value={1}>
                <Select
                  placeholder={placeholder || "Search"}
                  items={items}
                  value={value}
                  onChange={onChange}
                  multiple={multiple}
                  style={{ backgroundColor: "black" }}
                  renderRow={(item, isSelected, toggleItem) =>
                renderRow(item, isSelected, toggleItem)
              }
                  filterItem={filterItem}
                />
            </Fade>
        )}
            <FormGroup>
                <Column>
                    <Button style={Styles.mb5} onPress={this.onDone}>
                        Done
                    </Button>
                </Column>
            </FormGroup>
        </Flex>
    );
  }
};

module.exports = NativeModal;
