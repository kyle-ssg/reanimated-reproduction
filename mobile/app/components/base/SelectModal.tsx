import React, { useState } from 'react'
import { Loader } from './Loader'
import Select from './forms/Select'
import FormGroup from './grid/FormGroup'
import Column from './grid/Column'
import Flex from './grid/Flex'
import Button from './forms/Button'

interface Props {
  componentId?: string
  value?: {}
  isLoading?: boolean
  multiple?: boolean
  items?: any
  onChange: (value: {}) => void
  placeholder?: string
  filterItem?: () => void
  renderRow?: (item, isSelected, toggleItem) => void
  autoclose?: boolean
}

const NativeModal: React.FC<Props> = ({
  autoclose,
  value,
  isLoading,
  multiple,
  items,
  placeholder,
  filterItem,
  renderRow,
  onChange,
}) => {
  const [modalStatus, setModalStatus] = useState({
    isLoading: true,
    value: value,
  })

  const onDone = () => onChange(modalStatus.value)

  const onChangeHandler = (value: {}) => {
    if (autoclose) {
      onChange(value)
    } else {
      setModalStatus((prevState) => ({
        ...prevState,
        value: value,
      }))
    }
  }

  return (
    <Flex style={[Styles.body]}>
      {isLoading && (
        <Flex style={Styles.centeredContainer}>
          <Loader />
        </Flex>
      )}
      {items && (
        <Fade style={{ flex: 1 }} autostart value={1}>
          <Select
            placeholder={placeholder || 'Search'}
            items={items}
            value={value}
            onChange={onChangeHandler}
            multiple={multiple}
            style={{ backgroundColor: 'black' }}
            renderRow={(item, isSelected, toggleItem) =>
              renderRow(item, isSelected, toggleItem)
            }
            filterItem={filterItem}
          />
        </Fade>
      )}
      <FormGroup>
        <Column>
          <Button style={Styles.mb5} onPress={onDone}>
            Done
          </Button>
        </Column>
      </FormGroup>
    </Flex>
  )
}

export default NativeModal
