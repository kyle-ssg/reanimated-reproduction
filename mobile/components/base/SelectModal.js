//propTypes: uri: RequiredString
const NativeModal = class extends React.Component {
	constructor(props) {
		super(props);
		this.props.navigator.setOnNavigatorEvent(this.onNavigatorEvent.bind(this));
		this.state = {isLoading: true, value: props.value}
	}

	onNavigatorEvent(event) {
		if (event.id == 'close') {
			this.props.navigator.dismissModal();
		} else if (event.id == Constants.navEvents.SHOW) {

		} else {
		}

	}

	onDone = () => {
		this.props.navigator.dismissModal();
		this.props.onChange(this.state.value);
	};

	render() {
		const {isLoading, multiple, items} = this.props;
		const {value} = this.state;
		return (
			<Flex style={[Styles.body]}>
				{isLoading && <Flex style={Styles.centeredContainer}><Loader/></Flex>}
				{items && <Fade style={{flex: 1}} autostart={true} value={1}>
					<Select
						placeholder={this.props.placeholder || "Search"}
						items={items}
						value={value}
						onChange={(value) => this.setState({value})}
						multiple={multiple}
						renderRow={(item, isSelected, toggleItem) => this.props.renderRow(item, isSelected, toggleItem)}
						filterItem={this.props.filterItem}
					/>
				</Fade>}
				<FormGroup>
					<Column>
						<Button onPress={this.onDone}>Done</Button>
					</Column>
				</FormGroup>
			</Flex>
		);
	}
};

module.exports = NativeModal;
