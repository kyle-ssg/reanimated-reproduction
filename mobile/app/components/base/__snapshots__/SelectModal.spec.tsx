import * as React from 'react'
import SelectModal from '../SelectModal'
import renderer from 'react-test-renderer'

describe('<SelectModal />', () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(<SelectModal />)
    expect(tree).toMatchSnapshot()
  })
})
