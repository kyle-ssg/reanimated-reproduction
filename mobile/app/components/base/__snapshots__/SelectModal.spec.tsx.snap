// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectModal /> renders correctly 1`] = `
<View
  style={
    Array [
      Array [
        Object {
          "flex": 1,
        },
      ],
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "paddingBottom": 8,
          "paddingTop": 8,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "marginLeft": 10,
            "marginRight": 10,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
              "position": "relative",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#1AC0C6",
                "borderRadius": undefined,
                "height": 48,
                "justifyContent": "center",
                "overflow": "hidden",
              },
              Object {
                "marginBottom": 8,
              },
              null,
            ]
          }
        >
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontFamily": "System",
              }
            }
          >
            Done
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
